File: Auto/car_app_library/showcase/src/main/java/com/google/android/libraries/car/app/samples/showcase/navigation/PlaceListNavigationTemplateDemoScreen.java
/*
 * Copyright (C) 2020 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.google.android.libraries.car.app.samples.showcase.navigation;

import androidx.annotation.NonNull;
import com.google.android.libraries.car.app.CarContext;
import com.google.android.libraries.car.app.Screen;
import com.google.android.libraries.car.app.model.Action;
import com.google.android.libraries.car.app.model.ActionStrip;
import com.google.android.libraries.car.app.model.Template;
import com.google.android.libraries.car.app.navigation.model.PlaceListNavigationTemplate;
import com.google.android.libraries.car.app.samples.showcase.common.SamplePlaces;

/** Creates a screen using the {@link PlaceListNavigationTemplate} */
public final class PlaceListNavigationTemplateDemoScreen extends Screen {
  private SamplePlaces mPlaces;

  public PlaceListNavigationTemplateDemoScreen(CarContext carContext) {
    super(carContext);
    mPlaces = SamplePlaces.create(this);
  }

  @NonNull
  @Override
  public Template getTemplate() {
    return PlaceListNavigationTemplate.builder()
        .setItemList(mPlaces.getPlaceList())
        .setTitle("Place List Navigation Template Demo")
        .setHeaderAction(Action.BACK)
        .setActionStrip(
            ActionStrip.builder()
                .addAction(Action.builder().setTitle("Search").setOnClickListener(() -> {}).build())
                .build())
        .build();
  }
}

File: car_app_library/showcase/shared/src/main/java/androidx/car/app/sample/showcase/common/navigation/PlaceListNavigationTemplateDemoScreen.kt
/*
 * Copyright (C) 2021 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package androidx.car.app.sample.showcase.common.navigation

import androidx.car.app.CarContext
import androidx.car.app.CarToast
import androidx.car.app.Screen
import androidx.car.app.model.Action
import androidx.car.app.model.ActionStrip
import androidx.car.app.model.CarIcon
import androidx.car.app.model.Header
import androidx.car.app.model.Template
import androidx.car.app.navigation.model.PlaceListNavigationTemplate
import androidx.car.app.sample.showcase.common.R
import androidx.car.app.sample.showcase.common.common.SamplePlaces.Companion.create
import androidx.car.app.sample.showcase.common.navigation.routing.RoutingDemoModels
import androidx.core.graphics.drawable.IconCompat

/** Creates a screen using the [PlaceListNavigationTemplate]  */
class PlaceListNavigationTemplateDemoScreen(carContext: CarContext) : Screen(carContext) {
    private val mPlaces = create(this)

    private var mIsFavorite = false

    override fun onGetTemplate(): Template {
        val header = Header.Builder()
            .setStartHeaderAction(Action.BACK)
            .addEndHeaderAction(
                Action.Builder()
                    .setIcon(
                        CarIcon.Builder(
                            IconCompat.createWithResource(
                                carContext,
                                if (mIsFavorite
                                ) R.drawable.ic_favorite_filled_white_24dp
                                else R.drawable.ic_favorite_white_24dp
                            )
                        )
                            .build()
                    )
                    .setOnClickListener {
                        CarToast.makeText(
                            carContext,
                            if (mIsFavorite
                            ) carContext
                                .getString(R.string.favorite_toast_msg)
                            else carContext.getString(
                                R.string.not_favorite_toast_msg
                            ),
                            CarToast.LENGTH_SHORT
                        )
                            .show()
                        mIsFavorite = !mIsFavorite
                        invalidate()
                    }
                    .build())
            .addEndHeaderAction(
                Action.Builder()
                    .setOnClickListener { finish() }
                    .setIcon(
                        CarIcon.Builder(
                            IconCompat.createWithResource(
                                carContext,
                                R.drawable.ic_close_white_24dp
                            )
                        )
                            .build()
                    )
                    .build())
            .setTitle(carContext.getString(R.string.place_list_nav_template_demo_title))
            .build()

        return PlaceListNavigationTemplate.Builder()
            .setItemList(mPlaces.placeList)
            .setHeader(header)
            .setMapActionStrip(RoutingDemoModels.getMapActionStrip(carContext))
            .setActionStrip(
                ActionStrip.Builder()
                    .addAction(
                        Action.Builder()
                            .setTitle(
                                carContext.getString(
                                    R.string.search_action_title
                                )
                            )
                            .setOnClickListener {}
                            .build())
                    .build())
            .build()
    }
}
